### PR TITLE
Updated Editor Actions Location Settings and UI Changes

### PR DESCRIPTION
**Pull request by benibenj**
This pull request includes updates to the editor actions location settings and related UI changes. The editor actions can now be set to display in the default location, the title bar, or hidden entirely. This provides more flexibility to the user in customizing their editor interface. The UI has also been updated to reflect these changes.

### CODE REVIEW

File: src/vs/workbench/browser/parts/editor/editor.contribution.ts [hunk 1]
```diff
@@ -357,11 +357,18 @@ MenuRegistry.appendMenuItem(MenuId.EditorTabsBarContext, { command: { id: SPLIT_
...
+
 MenuRegistry.appendMenuItem(MenuId.EditorTabsBarContext, { submenu: MenuId.EditorTabsBarShowTabsSubmenu, title: localize('tabBar', "Tab Bar"), group: '3_config', order: 10 });
 MenuRegistry.appendMenuItem(MenuId.EditorTabsBarShowTabsSubmenu, { command: { id: ShowMultipleEditorTabsAction.ID, title: localize('multipleTabs', "Multiple Tabs"), toggled: ContextKeyExpr.equals('config.workbench.editor.showTabs', 'multiple') }, group: '1_config', order: 10 });
 MenuRegistry.appendMenuItem(MenuId.EditorTabsBarShowTabsSubmenu, { command: { id: ShowSingleEditorTabAction.ID, title: localize('singleTab', "Single Tab"), toggled: ContextKeyExpr.equals('config.workbench.editor.showTabs', 'single') }, group: '1_config', order: 20 });
 MenuRegistry.appendMenuItem(MenuId.EditorTabsBarShowTabsSubmenu, { command: { id: HideEditorTabsAction.ID, title: localize('hideTabBar', "Hide"), toggled: ContextKeyExpr.equals('config.workbench.editor.showTabs', 'none') }, group: '1_config', order: 30 });
 
+// Comment: If native title bar then toggle hidden and remove title bar menu item, If no tab bar then toggle title bar and remove tab bar menu item
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/editor.contribution.ts:366]: This reads like a todo from you? If not, I would remove the comment.

File: src/vs/workbench/browser/actions/layoutActions.ts [hunk 2]
```diff
@@ -534,6 +534,99 @@ MenuRegistry.appendMenuItem(MenuId.MenubarAppearanceMenu, {
...
+
+export class HideEditorActionsAction extends Action2 {
+
+	static readonly ID = 'workbench.action.editorActionsHidden';
+
+	constructor() {
+		super({
+			id: HideEditorActionsAction.ID,
+			title: {
+				value: localize('hideEditorActons', "Hide Editor Actions"),
```
Comment (bpasero) [inline src/vs/workbench/browser/actions/layoutActions.ts:605]: Should we have a related "Show Editor Actions"? I know we have the "Move..." commands but maybe people would expect to search for "Show...".

File: src/vs/workbench/browser/parts/editor/editorTabsControl.ts [hunk 3]
```diff
@@ -96,8 +96,10 @@ export abstract class EditorTabsControl extends Themable implements IEditorTabsC
 		compact: 22 as const
 	};
 
+	protected editorToolbarContainer: HTMLElement | undefined;
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/editorTabsControl.ts:99]: Maybe `editorActionsToolbarContainer` to be consistent with names.

File: src/vs/workbench/browser/parts/editor/editorTabsControl.ts [hunk 4]
```diff
@@ -202,6 +234,10 @@ export abstract class EditorTabsControl extends Themable implements IEditorTabsC
 	}
 
 	protected updateEditorActionsToolbar(): void {
+		if (!this.editorActionsEnabled) {
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/editorTabsControl.ts:237]: There is another method below `protected clearEditorActionsToolbar(): void {` which I wonder how we should deal with it based on the setting. What does it mean to "clear" the editor actions toolbar, when it is rooted in the title?

**Review by bpasero — CHANGES_REQUESTED** [review]
Nicely done, only minor feedback.

File: src/vs/workbench/browser/actions/layoutActions.ts (hunk 2 above)
Comment (benibenj) [reply src/vs/workbench/browser/actions/layoutActions.ts:605]: I agree, that makes sense

File: src/vs/workbench/browser/parts/editor/editorTabsControl.ts (hunk 4 above)
Comment (benibenj) [reply src/vs/workbench/browser/parts/editor/editorTabsControl.ts:237]: If clear is called and the actions are in the title bar then `this.editorActionsToolbar` will be undefined and nothing will and should happen. I'll write it a bit more explicit.

`clearEditorActionsToolbar()` is the same as `updateEditorActionsToolbar()` just a bit faster when knowing there are no editor actions.

**Review by bpasero — APPROVED** [review]
Nicely done, only minor feedback.

**Pull request merged**